assets/src/domain/eventEditor/services/apollo/mutations/datetimes/useReorderDatetimes.ts

```diff
@@ -0,0 +1,60 @@
+import { useCallback } from 'react';
+
+import { Datetime } from '../../types';
+import useReorderEntities from '../useReorderEntities';
+import { EntityTableProps } from '@appLayout/entityList';
+import { DatetimesFilterStateManager as DFSM } from '@edtrServices/filterState';
+import { useDatetimes, useDatetimeQueryOptions } from '@edtrServices/apollo/queries';
+import { useUpdateDatetimeList } from '@edtrHooks/index';
+
+type SortResponder = EntityTableProps<Datetime, DFSM>['onSort'];
+
+interface ReorderDatetimes {
+	sortResponder: SortResponder;
+}
+
+const useReorderDatetimes = (filteredEntities: Array<Datetime>): ReorderDatetimes => {
+	const { sortEntities } = useReorderEntities<Datetime>({ entityType: 'DATETIME' });
+	const allEntities = useDatetimes();
+	const queryOptions = useDatetimeQueryOptions();
+	const updateDatetimeList = useUpdateDatetimeList();
+
+	const updateEntityList = useCallback(
+		(updatedEntities) => {
+			updateDatetimeList({
+				...queryOptions,
+				data: {
+					espressoDatetimes: {
+						nodes: updatedEntities,
+						__typename: 'EspressoRootQueryDatetimesConnection',
+					},
+				},
+			});
+		},
```

Comment:
this looks like a Christmas tree :christmas_tree: 
can this somehow be flattened or abstracted?

Reply:
Done